### ISSUE TITLE
Request takes too long to respond

### ISSUE THREAD

**Issue by Fradd747** [allegro/allegro-api#10440]
Hey, when I call allegro endpoint "https://api.allegro.pl/sale/products?mode=GTIN&phrase=7320545205542&language=cs-CZ", first approx. 20 calls are returned in no time and then it takes almost 20 seconds to respond. Only Allegro API limits I found says 9000 requests per minute.

Can you tell me what are the limits?

**Comment by MaciejFrackowiak** [comment]
Yes, the basic limit is 9000 requests per minute. Also, selected resources have an additional mechanism which limits requests - a leaky bucket. Please, read about it [here](https://developer.allegro.pl/tutorials/basic-information-VL6YelvVKTn#limiting-the-number-of-queries-limits).

**Comment by Fradd747** [comment]
You're totally right. I have also read that. 

Can we look into this particular endpoint I sent earlier (sale/products search)? The documentation says "This resource is limited with Leaky Bucket mechanism" but I haven't found what is the limit.

**Comment by MaciejFrackowiak** [comment]
We don't inform about it - the limit adjusts to the number of requests made by the user.

**Comment by Fradd747** [comment]
Okay, so how should I check for example if 10k products exist in Allegro according to their EAN if I don't know the rate limit for the method and sending requests sequentially will take a decade?

**Comment by MaciejFrackowiak** [comment]
You need to stagger your requests to avoid hitting the mentioned limitation.
